### ISSUE TITLE
`cmd` process type cannot be overridden in a Procfile

### ISSUE THREAD

**Issue by bacongobbler** [deis/deis#2668]
This may be unintentional behaviour, so I wanted to point this out before I fix to have a discussion.

When you use a Dockerfile or a Docker image, the default process type is `cmd`. `cmd` is defined as the default command from the docker image (the CMD entry). This cannot be overridden due to how we handle the command logic: https://github.com/deis/deis/blob/e592528f2c04af27400c204632da304c17066737/controller/api/models.py#L403-L406

Basically it will always return `''`, even if you defined a `cmd` process type in a Procfile. We should allow users to override the default `cmd` process type in a Procfile if they so desire.
